In [1]:
#configuración en google colab
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [2]:
! pip install stargazer -q
! pip install linearmodels -q

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 19.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.2/94.2 kB 8.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.7/43.7 kB 3.2 MB/s eta 0:00:00


In [3]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
from stargazer.stargazer import Stargazer
from linearmodels.panel import PanelOLS, RandomEffects, PooledOLS

In [4]:
# directorios (path):
path = "gdrive/MyDrive/my_notebooks/datasets/"

In [5]:
# Cargar información con rezagos
df = pd.read_pickle(path + 'df_rezagos.pkl')
df

Ano  Semana    Zona        Finca  municipio  Semana_seq  Ano_Semana  \
0    2021      24  CENTRO  MADRE SELVA     CAREPA           1      202124   
1    2021      25  CENTRO  MADRE SELVA     CAREPA           2      202125   
2    2021      26  CENTRO  MADRE SELVA     CAREPA           3      202126   
3    2021      27  CENTRO  MADRE SELVA     CAREPA           4      202127   
4    2021      28  CENTRO  MADRE SELVA     CAREPA           5      202128   
..    ...     ...     ...          ...        ...         ...         ...   
448  2024      14     SUR      CORALES  CHIGORODÓ         147      202414   
449  2024      15     SUR      CORALES  CHIGORODÓ         148      202415   
450  2024      16     SUR      CORALES  CHIGORODÓ         149      202416   
451  2024      17     SUR      CORALES  CHIGORODÓ         150      202417   
452  2024      18     SUR      CORALES  CHIGORODÓ         151      202418   

     temp_media_lag1  temp_media_lag2  temp_media_lag3  ...  \
0                NaN              NaN              NaN  ...   
1              27.89              NaN              NaN  ...   
2              27.71            27.89              NaN  ...   
3              26.44            27.71            27.89  ...   
4              27.34            26.44            27.71  ...   
..               ...              ...              ...  ...   
448            28.49            29.01            27.60  ...   
449            28.18            28.49            29.01  ...   
450            28.27            28.18            28.49  ...   
451            27.87            28.27            28.18  ...   
452            28.52            27.87            28.27  ...   

     Velocidad_del_viento_lag43  Velocidad_del_viento_lag44  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.3                         0.1   
449                         0.2                         0.3   
450                         0.8                         0.2   
451                         0.4                         0.8   
452                         1.0                         0.4   

     Velocidad_del_viento_lag45  Velocidad_del_viento_lag46  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.2                         0.5   
449                         0.1                         0.2   
450                         0.3                         0.1   
451                         0.2                         0.3   
452                         0.8                         0.2   

     Velocidad_del_viento_lag47  Velocidad_del_viento_lag48  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.2                         0.3   
449                         0.5                         0.2   
450                         0.2                         0.5   
451                         0.1                         0.2   
452                         0.3                         0.1   

     Velocidad_del_viento_lag49  Velocidad_del_viento

In [6]:
# Generar un identificador por Zona
df.loc[df["Zona"] == "NORTE", "id"] = 1
df.loc[df["Zona"] == "CENTRO", "id"] = 2
df.loc[df["Zona"] == "SUR", "id"] = 3
df

Ano  Semana    Zona        Finca  municipio  Semana_seq  Ano_Semana  \
0    2021      24  CENTRO  MADRE SELVA     CAREPA           1      202124   
1    2021      25  CENTRO  MADRE SELVA     CAREPA           2      202125   
2    2021      26  CENTRO  MADRE SELVA     CAREPA           3      202126   
3    2021      27  CENTRO  MADRE SELVA     CAREPA           4      202127   
4    2021      28  CENTRO  MADRE SELVA     CAREPA           5      202128   
..    ...     ...     ...          ...        ...         ...         ...   
448  2024      14     SUR      CORALES  CHIGORODÓ         147      202414   
449  2024      15     SUR      CORALES  CHIGORODÓ         148      202415   
450  2024      16     SUR      CORALES  CHIGORODÓ         149      202416   
451  2024      17     SUR      CORALES  CHIGORODÓ         150      202417   
452  2024      18     SUR      CORALES  CHIGORODÓ         151      202418   

     temp_media_lag1  temp_media_lag2  temp_media_lag3  ...  \
0                NaN              NaN              NaN  ...   
1              27.89              NaN              NaN  ...   
2              27.71            27.89              NaN  ...   
3              26.44            27.71            27.89  ...   
4              27.34            26.44            27.71  ...   
..               ...              ...              ...  ...   
448            28.49            29.01            27.60  ...   
449            28.18            28.49            29.01  ...   
450            28.27            28.18            28.49  ...   
451            27.87            28.27            28.18  ...   
452            28.52            27.87            28.27  ...   

     Velocidad_del_viento_lag44  Velocidad_del_viento_lag45  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.1                         0.2   
449                         0.3                         0.1   
450                         0.2                         0.3   
451                         0.8                         0.2   
452                         0.4                         0.8   

     Velocidad_del_viento_lag46  Velocidad_del_viento_lag47  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.5                         0.2   
449                         0.2                         0.5   
450                         0.1                         0.2   
451                         0.3                         0.1   
452                         0.2                         0.3   

     Velocidad_del_viento_lag48  Velocidad_del_viento_lag49  \
0                           NaN                         NaN   
1                           NaN                         NaN   
2                           NaN                         NaN   
3                           NaN                         NaN   
4                           NaN                         NaN   
..                          ...                         ...   
448                         0.3                         0.2   
449                         0.2                         0.3   
450                         0.5                         0.2   
451                         0.2                         0.5   
452                         0.1                         0.2   

     Velocidad_del_viento_lag50  Velocidad_del_viento

Generar multi-indice con 'id' y 'Semana_seq', de forma que contenga una llave para ordenamiento de tiempo

In [7]:
df.set_index(["id", "Semana_seq"], inplace=True)
df

Ano  Semana    Zona        Finca  municipio  Ano_Semana  \
id  Semana_seq                                                             
2.0 1           2021      24  CENTRO  MADRE SELVA     CAREPA      202124   
    2           2021      25  CENTRO  MADRE SELVA     CAREPA      202125   
    3           2021      26  CENTRO  MADRE SELVA     CAREPA      202126   
    4           2021      27  CENTRO  MADRE SELVA     CAREPA      202127   
    5           2021      28  CENTRO  MADRE SELVA     CAREPA      202128   
...              ...     ...     ...          ...        ...         ...   
3.0 147         2024      14     SUR      CORALES  CHIGORODÓ      202414   
    148         2024      15     SUR      CORALES  CHIGORODÓ      202415   
    149         2024      16     SUR      CORALES  CHIGORODÓ      202416   
    150         2024      17     SUR      CORALES  CHIGORODÓ      202417   
    151         2024      18     SUR      CORALES  CHIGORODÓ      202418   

                temp_media_lag1  temp_media_lag2  temp_media_lag3  \
id  Semana_seq                                                      
2.0 1                       NaN              NaN              NaN   
    2                     27.89              NaN              NaN   
    3                     27.71            27.89              NaN   
    4                     26.44            27.71            27.89   
    5                     27.34            26.44            27.71   
...                         ...              ...              ...   
3.0 147                   28.49            29.01            27.60   
    148                   28.18            28.49            29.01   
    149                   28.27            28.18            28.49   
    150                   27.87            28.27            28.18   
    151                   28.52            27.87            28.27   

                temp_media_lag4  ...  Velocidad_del_viento_lag43  \
id  Semana_seq                   ...                               
2.0 1                       NaN  ...                         NaN   
    2                       NaN  ...                         NaN   
    3                       NaN  ...                         NaN   
    4                       NaN  ...                         NaN   
    5                     27.89  ...                         NaN   
...                         ...  ...                         ...   
3.0 147                   28.01  ...                         0.3   
    148                   27.60  ...                         0.2   
    149                   29.01  ...                         0.8   
    150                   28.49  ...                         0.4   
    151                   28.18  ...                         1.0   

                Velocidad_del_viento_lag44  Velocidad_del_viento_lag45  \
id  Semana_seq                                                           
2.0 1                                  NaN                         NaN   
    2                                  NaN                         NaN   
    3                                  NaN                         NaN   
    4                                  NaN                         NaN   
    5                                  NaN                         NaN   
...                                    ...                         ...   
3.0 147                                0.1                         0.2   
    148                                0.3                         0.1   
    149                                0.2                         0.3   
    150                                0.8                         0.2   
    151                                0.4                         0.8   

                Velocidad_del_viento_lag46  Velocidad_del_viento_lag47  \
id  Semana_seq                                                           
2.0 1                                  NaN                         NaN   
    2                                  NaN                  

## Realizar prueba de hausman para determinar tipos de efecto en el dataframe completo

In [8]:
def calculate_aic_bic_hqic(res):
  import numpy as np
  k = len(res.params)
  llf = res.loglik
  aic = 2 * k - 2 * llf
  bic = np.log(res.nobs) * k - 2 * llf
  hqic = 2 * k * np.log(np.log(res.nobs)) - 2 * llf
  return aic, bic, hqic

In [9]:
def hausman(fe, re):
  from scipy import stats
  b = fe.params
  B = re.params
  v_b = fe.cov
  v_B = re.cov
  df = b[np.abs(b) < 1e8].size
  chi2 = np.dot((b - B).T, np.linalg.inv(v_b - v_B).dot(b - B))
  pval = stats.chi2.sf(chi2, df)
  return chi2, df, pval

Crear dataframe para almacenar los resultados de los modelos para cada # de rezago

In [10]:
# lista de regresiones lineales
lista_modelos = pd.DataFrame(columns=['variables_predictoras', 'modelo', 'tipo_efectos'])
lista_modelos

Empty DataFrame
Columns: [variables_predictoras, modelo, tipo_efectos]
Index: []

Crear lista con todas las variables rezagadas para realizar la selección entre ellas

In [11]:
variables_predictoras = []
for i in range(1, 52):  # Desde 1 hasta 51
  variables_predictoras.extend([f"temp_media_lag{i}", f"Precipitacion_lag{i}",
                                f"Humedad_relativa_lag{i}", f"Velocidad_del_viento_lag{i}"])

In [12]:
models = []  # Lista para almacenar los resultados del modelo

for i in range(1, 52):  # Desde 1 hasta 51
  exog_data = df[variables_predictoras[:i*4]]  # Seleccionar las primeras i*4 variables
  exog_data = sm.add_constant(exog_data)  # Agregar la constante

  # Fit the models
  #pool = PooledOLS(data.Peso_total_del_racimo_kg, exog_data)
  #pool_res = pool.fit()
  #print(pool_res)

  aleatorios = RandomEffects(df['Peso_total_del_racimo_kg'], exog_data)
  al_res = aleatorios.fit()

  fijos = PanelOLS(df['Peso_total_del_racimo_kg'], exog_data, entity_effects=True)
  fe_res = fijos.fit()

  aic_fe, bic_fe, hqic_fe = calculate_aic_bic_hqic(fe_res)
  aic_al, bic_al, hqic_al = calculate_aic_bic_hqic(al_res)

  hausman_results = hausman(fe_res, al_res)

  chi2 = hausman_results[0]
  pval = hausman_results[2]

  if chi2 < 0:
    print('chi2:', chi2)
    print("Efectos fijos son preferibles.")
    print(fe_res)
    lista_modelos.loc[len(lista_modelos)] = [variables_predictoras[:i*4],fe_res, 'Efectos fijos']
    rmse_fe = np.sqrt(np.mean(fe_res.resids ** 2))
    mae_fe = np.mean(np.abs(fe_res.resids))
    stargazer = Stargazer([fe_res])
    stargazer.add_line("RMSE", [f"{rmse_fe:.4f}"])
    stargazer.add_line("MAE", [f"{mae_fe:.4f}"])
    stargazer.add_line("AIC", [f"{aic_fe:.4f}"])
    stargazer.add_line("BIC", [f"{bic_fe:.4f}"])
    stargazer.add_line("HQIC", [f"{hqic_fe:.4f}"])
    models.append({"name": f"Modelo {i}", "result": fe_res, "aic": aic_al, "bic": bic_al, "hqic": hqic_al})

  elif pval < 0.05:
    print("Efectos fijos son preferibles (p-valor < 0.05).")
    print(fe_res)
    lista_modelos.loc[len(lista_modelos)] = [variables_predictoras[:i*4], fe_res, 'Efectos fijos']
    rmse_fe = np.sqrt(np.mean(fe_res.resids ** 2))
    mae_fe = np.mean(np.abs(fe_res.resids))
    stargazer = Stargazer([fe_res])
    stargazer.add_line("RMSE", [f"{rmse_fe:.4f}"])
    stargazer.add_line("MAE", [f"{mae_fe:.4f}"])
    stargazer.add_line("AIC", [f"{aic_fe:.4f}"])
    stargazer.add_line("BIC", [f"{bic_fe:.4f}"])
    stargazer.add_line("HQIC", [f"{hqic_fe:.4f}"])
    models.append({"name": f"Modelo {i}", "result": al_res, "aic": aic_al, "bic": bic_al, "hqic": hqic_al})
  else:
    print("Efectos aleatorios son preferibles.")
    print(al_res)
    lista_modelos.loc[len(lista_modelos)] = [variables_predictoras[:i*4], al_res, 'Efectos Aleatorios']
    rmse_al = np.sqrt(np.mean(al_res.resids ** 2))
    mae_al = np.mean(np.abs(al_res.resids))
    stargazer = Stargazer([al_res])
    stargazer.add_line("RMSE", [f"{rmse_al:.4f}"])
    stargazer.add_line("MAE", [f"{mae_al:.4f}"])
    stargazer.add_line("AIC", [f"{aic_al:.4f}"])
    stargazer.add_line("BIC", [f"{bic_al:.4f}"])
    stargazer.add_line("HQIC", [f"{hqic_al:.4f}"])
    models.append({"name": f"Modelo {i}", "result": al_res, "aic": aic_al, "bic": bic_al, "hqic": hqic_al})


/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -101.17072314953589
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.0792
Estimator:                         PanelOLS   R-squared (Between):             -0.0181
No. Observations:                       450   R-squared (Within):               0.0792
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0086
Time:                              23:10:04   Log-likelihood                   -1019.9
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      9.5206
Entities:                                 3   P-value                           0.0000
Avg Obs:                             150.00   Distribution:                   F(4,443)
Min Obs:                             150.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -118.30211582876387
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.1160
Estimator:                         PanelOLS   R-squared (Between):             -0.0124
No. Observations:                       447   R-squared (Within):               0.1160
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0230
Time:                              23:10:05   Log-likelihood                   -1005.4
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      7.1529
Entities:                                 3   P-value                           0.0000
Avg Obs:                             149.00   Distribution:                   F(8,436)
Min Obs:                             149.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -134.7573455085535
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.1363
Estimator:                         PanelOLS   R-squared (Between):             -0.0109
No. Observations:                       444   R-squared (Within):               0.1363
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0299
Time:                              23:10:05   Log-likelihood                   -994.99
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.6414
Entities:                                 3   P-value                           0.0000
Avg Obs:                             148.00   Distribution:                  F(12,429)
Min Obs:                             148.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -151.48954530752658
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.1519
Estimator:                         PanelOLS   R-squared (Between):             -0.0153
No. Observations:                       441   R-squared (Within):               0.1519
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0312
Time:                              23:10:06   Log-likelihood                   -985.71
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.7226
Entities:                                 3   P-value                           0.0000
Avg Obs:                             147.00   Distribution:                  F(16,422)
Min Obs:                             147.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -168.1830421551763
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.1753
Estimator:                         PanelOLS   R-squared (Between):             -0.0159
No. Observations:                       438   R-squared (Within):               0.1753
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0375
Time:                              23:10:07   Log-likelihood                   -974.07
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.4112
Entities:                                 3   P-value                           0.0000
Avg Obs:                             146.00   Distribution:                  F(20,415)
Min Obs:                             146.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -178.7810390130996
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.1956
Estimator:                         PanelOLS   R-squared (Between):             -0.0151
No. Observations:                       435   R-squared (Within):               0.1956
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0438
Time:                              23:10:08   Log-likelihood                   -962.52
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.1338
Entities:                                 3   P-value                           0.0000
Avg Obs:                             145.00   Distribution:                  F(24,408)
Min Obs:                             145.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -186.1712050377331
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.2108
Estimator:                         PanelOLS   R-squared (Between):             -0.0120
No. Observations:                       432   R-squared (Within):               0.2108
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0503
Time:                              23:10:09   Log-likelihood                   -951.78
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.8258
Entities:                                 3   P-value                           0.0000
Avg Obs:                             144.00   Distribution:                  F(28,401)
Min Obs:                             144.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -193.84086818925738
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.2288
Estimator:                         PanelOLS   R-squared (Between):             -0.0120
No. Observations:                       429   R-squared (Within):               0.2288
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0550
Time:                              23:10:10   Log-likelihood                   -939.54
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.6519
Entities:                                 3   P-value                           0.0000
Avg Obs:                             143.00   Distribution:                  F(32,394)
Min Obs:                             143.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -200.32829724131122
Efectos fijos son preferibles.


/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.2456
Estimator:                         PanelOLS   R-squared (Between):             -0.0143
No. Observations:                       426   R-squared (Within):               0.2456
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0577
Time:                              23:10:12   Log-likelihood                   -926.49
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.5000
Entities:                                 3   P-value                           0.0000
Avg Obs:                             142.00   Distribution:                  F(36,387)
Min Obs:                             142.00                                           
Max Obs:                             142.00

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -213.46936777920195
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.2799
Estimator:                         PanelOLS   R-squared (Between):             -0.0167
No. Observations:                       420   R-squared (Within):               0.2799
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0648
Time:                              23:10:14   Log-likelihood                   -900.94
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.2954
Entities:                                 3   P-value                           0.0000
Avg Obs:                             140.00   Distribution:                  F(44,373)
Min Obs:                             140.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -219.83969815241682
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.2962
Estimator:                         PanelOLS   R-squared (Between):             -0.0169
No. Observations:                       417   R-squared (Within):               0.2962
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0691
Time:                              23:10:15   Log-likelihood                   -888.79
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.2086
Entities:                                 3   P-value                           0.0000
Avg Obs:                             139.00   Distribution:                  F(48,366)
Min Obs:                             139.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -225.18153658138147
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3096
Estimator:                         PanelOLS   R-squared (Between):             -0.0172
No. Observations:                       414   R-squared (Within):               0.3096
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0725
Time:                              23:10:17   Log-likelihood                   -877.63
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.0958
Entities:                                 3   P-value                           0.0000
Avg Obs:                             138.00   Distribution:                  F(52,359)
Min Obs:                             138.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -231.90994249882908
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3196
Estimator:                         PanelOLS   R-squared (Between):             -0.0169
No. Observations:                       411   R-squared (Within):               0.3196
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0753
Time:                              23:10:18   Log-likelihood                   -866.50
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.9532
Entities:                                 3   P-value                           0.0000
Avg Obs:                             137.00   Distribution:                  F(56,352)
Min Obs:                             137.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -238.919137984732
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3354
Estimator:                         PanelOLS   R-squared (Between):             -0.0152
No. Observations:                       408   R-squared (Within):               0.3354
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0805
Time:                              23:10:19   Log-likelihood                   -854.19
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.9012
Entities:                                 3   P-value                           0.0000
Avg Obs:                             136.00   Distribution:                  F(60,345)
Min Obs:                             136.00                                

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -255.62125462517616
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3493
Estimator:                         PanelOLS   R-squared (Between):             -0.0133
No. Observations:                       405   R-squared (Within):               0.3493
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0857
Time:                              23:10:20   Log-likelihood                   -842.97
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.8355
Entities:                                 3   P-value                           0.0000
Avg Obs:                             135.00   Distribution:                  F(64,338)
Min Obs:                             135.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -277.71494577841
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3646
Estimator:                         PanelOLS   R-squared (Between):             -0.0166
No. Observations:                       402   R-squared (Within):               0.3646
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0877
Time:                              23:10:21   Log-likelihood                   -832.43
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.7936
Entities:                                 3   P-value                           0.0000
Avg Obs:                             134.00   Distribution:                  F(68,331)
Min Obs:                             134.00                                 

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -308.97674656022957
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3803
Estimator:                         PanelOLS   R-squared (Between):             -0.0076
No. Observations:                       399   R-squared (Within):               0.3803
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.0988
Time:                              23:10:23   Log-likelihood                   -821.25
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.7618
Entities:                                 3   P-value                           0.0000
Avg Obs:                             133.00   Distribution:                  F(72,324)
Min Obs:                             133.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -344.4108881760171
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.3942
Estimator:                         PanelOLS   R-squared (Between):              0.0014
No. Observations:                       396   R-squared (Within):               0.3942
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1096
Time:                              23:10:24   Log-likelihood                   -810.69
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.7147
Entities:                                 3   P-value                           0.0000
Avg Obs:                             132.00   Distribution:                  F(76,317)
Min Obs:                             132.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -382.17482861716286
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.4092
Estimator:                         PanelOLS   R-squared (Between):             -0.0015
No. Observations:                       393   R-squared (Within):               0.4092
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1107
Time:                              23:10:25   Log-likelihood                   -799.23
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.6834
Entities:                                 3   P-value                           0.0000
Avg Obs:                             131.00   Distribution:                  F(80,310)
Min Obs:                             131.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -416.53250627374615
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.4277
Estimator:                         PanelOLS   R-squared (Between):              0.0072
No. Observations:                       390   R-squared (Within):               0.4277
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1214
Time:                              23:10:27   Log-likelihood                   -786.79
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.6959
Entities:                                 3   P-value                           0.0000
Avg Obs:                             130.00   Distribution:                  F(84,303)
Min Obs:                             130.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -428.0542685296716
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.4590
Estimator:                         PanelOLS   R-squared (Between):              0.0230
No. Observations:                       387   R-squared (Within):               0.4590
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1405
Time:                              23:10:28   Log-likelihood                   -769.45
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      2.8543
Entities:                                 3   P-value                           0.0000
Avg Obs:                             129.00   Distribution:                  F(88,296)
Min Obs:                             129.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -427.65004692726353
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.4889
Estimator:                         PanelOLS   R-squared (Between):              0.0326
No. Observations:                       384   R-squared (Within):               0.4889
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1546
Time:                              23:10:29   Log-likelihood                   -752.05
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.0052
Entities:                                 3   P-value                           0.0000
Avg Obs:                             128.00   Distribution:                  F(92,289)
Min Obs:                             128.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -433.6688910001839
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.5217
Estimator:                         PanelOLS   R-squared (Between):              0.0311
No. Observations:                       381   R-squared (Within):               0.5217
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1611
Time:                              23:10:30   Log-likelihood                   -732.98
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.2039
Entities:                                 3   P-value                           0.0000
Avg Obs:                             127.00   Distribution:                  F(96,282)
Min Obs:                             127.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -456.4623286039815
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.5435
Estimator:                         PanelOLS   R-squared (Between):              0.0315
No. Observations:                       378   R-squared (Within):               0.5435
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1660
Time:                              23:10:31   Log-likelihood                   -717.79
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.2735
Entities:                                 3   P-value                           0.0000
Avg Obs:                             126.00   Distribution:                 F(100,275)
Min Obs:                             126.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -470.02047889078733
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.5747
Estimator:                         PanelOLS   R-squared (Between):              0.0148
No. Observations:                       375   R-squared (Within):               0.5747
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1607
Time:                              23:10:33   Log-likelihood                   -698.14
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.4825
Entities:                                 3   P-value                           0.0000
Avg Obs:                             125.00   Distribution:                 F(104,268)
Min Obs:                             125.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -466.08600529212504
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.6125
Estimator:                         PanelOLS   R-squared (Between):              0.0204
No. Observations:                       372   R-squared (Within):               0.6125
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1749
Time:                              23:10:34   Log-likelihood                   -675.65
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.8198
Entities:                                 3   P-value                           0.0000
Avg Obs:                             124.00   Distribution:                 F(108,261)
Min Obs:                             124.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -492.7699972096577
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.6484
Estimator:                         PanelOLS   R-squared (Between):              0.0227
No. Observations:                       369   R-squared (Within):               0.6484
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1865
Time:                              23:10:35   Log-likelihood                   -652.24
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.1815
Entities:                                 3   P-value                           0.0000
Avg Obs:                             123.00   Distribution:                 F(112,254)
Min Obs:                             123.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -543.5242294336607
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.6835
Estimator:                         PanelOLS   R-squared (Between):              0.0202
No. Observations:                       366   R-squared (Within):               0.6835
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1941
Time:                              23:10:35   Log-likelihood                   -628.06
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.5975
Entities:                                 3   P-value                           0.0000
Avg Obs:                             122.00   Distribution:                 F(116,247)
Min Obs:                             122.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -604.9324058432397
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7175
Estimator:                         PanelOLS   R-squared (Between):              0.0273
No. Observations:                       363   R-squared (Within):               0.7175
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2087
Time:                              23:10:36   Log-likelihood                   -602.84
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0803
Entities:                                 3   P-value                           0.0000
Avg Obs:                             121.00   Distribution:                 F(120,240)
Min Obs:                             121.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -679.8050299740735
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7403
Estimator:                         PanelOLS   R-squared (Between):              0.0168
No. Observations:                       360   R-squared (Within):               0.7403
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2070
Time:                              23:10:37   Log-likelihood                   -583.11
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.3557
Entities:                                 3   P-value                           0.0000
Avg Obs:                             120.00   Distribution:                 F(124,233)
Min Obs:                             120.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -852.165919412911
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7528
Estimator:                         PanelOLS   R-squared (Between):              0.0113
No. Observations:                       357   R-squared (Within):               0.7528
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2065
Time:                              23:10:38   Log-likelihood                   -569.88
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.3764
Entities:                                 3   P-value                           0.0000
Avg Obs:                             119.00   Distribution:                 F(128,226)
Min Obs:                             119.00                                

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -1096.4179012077675
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7647
Estimator:                         PanelOLS   R-squared (Between):              0.0050
No. Observations:                       354   R-squared (Within):               0.7647
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2053
Time:                              23:10:38   Log-likelihood                   -556.12
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.3933
Entities:                                 3   P-value                           0.0000
Avg Obs:                             118.00   Distribution:                 F(132,219)
Min Obs:                             118.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -1894.0452075123894
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7690
Estimator:                         PanelOLS   R-squared (Between):              0.0081
No. Observations:                       351   R-squared (Within):               0.7690
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2087
Time:                              23:10:39   Log-likelihood                   -548.11
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.1893
Entities:                                 3   P-value                           0.0000
Avg Obs:                             117.00   Distribution:                 F(136,212)
Min Obs:                             117.00                              

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -8384.95893812478
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7757
Estimator:                         PanelOLS   R-squared (Between):             -0.0153
No. Observations:                       348   R-squared (Within):               0.7757
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1932
Time:                              23:10:40   Log-likelihood                   -538.37
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0649
Entities:                                 3   P-value                           0.0000
Avg Obs:                             116.00   Distribution:                 F(140,205)
Min Obs:                             116.00                                

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7859
Estimator:                         PanelOLS   R-squared (Between):             -0.0274
No. Observations:                       345   R-squared (Within):               0.7859
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1868
Time:                              23:10:41   Log-likelihood                   -525.86
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0473
Entities:                                 3   P-value                           0.0000
Avg Obs:                             115.00   Distribution:                 F(144,198)
Min Obs:                             115.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7938
Estimator:                         PanelOLS   R-squared (Between):             -0.0161
No. Observations:                       342   R-squared (Within):               0.7938
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.1970
Time:                              23:10:42   Log-likelihood                   -514.67
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.9696
Entities:                                 3   P-value                           0.0000
Avg Obs:                             114.00   Distribution:                 F(148,191)
Min Obs:                             114.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8035
Estimator:                         PanelOLS   R-squared (Between):              0.0264
No. Observations:                       339   R-squared (Within):               0.8035
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2305
Time:                              23:10:43   Log-likelihood                   -501.67
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.9488
Entities:                                 3   P-value                           0.0000
Avg Obs:                             113.00   Distribution:                 F(152,184)
Min Obs:                             113.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8128
Estimator:                         PanelOLS   R-squared (Between):              0.0585
No. Observations:                       336   R-squared (Within):               0.8128
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2570
Time:                              23:10:44   Log-likelihood                   -489.61
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.9258
Entities:                                 3   P-value                           0.0000
Avg Obs:                             112.00   Distribution:                 F(156,177)
Min Obs:                             112.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8274
Estimator:                         PanelOLS   R-squared (Between):              0.0771
No. Observations:                       333   R-squared (Within):               0.8274
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.2755
Time:                              23:10:45   Log-likelihood                   -472.50
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0920
Entities:                                 3   P-value                           0.0000
Avg Obs:                             111.00   Distribution:                 F(160,170)
Min Obs:                             111.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8362
Estimator:                         PanelOLS   R-squared (Between):              0.1072
No. Observations:                       330   R-squared (Within):               0.8362
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.3006
Time:                              23:10:46   Log-likelihood                   -460.18
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0744
Entities:                                 3   P-value                           0.0000
Avg Obs:                             110.00   Distribution:                 F(164,163)
Min Obs:                             110.00                                       

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos aleatorios son preferibles.
                           RandomEffects Estimation Summary                           
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7639
Estimator:                    RandomEffects   R-squared (Between):              0.8575
No. Observations:                       327   R-squared (Within):               0.5058
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.7639
Time:                              23:10:47   Log-likelihood                   -732.33
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.0435
Entities:                                 3   P-value                           0.0000
Avg Obs:                             109.00   Distribution:                 F(168,158)
Min Obs:                             109.00                                           
Max Obs

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos aleatorios son preferibles.
                           RandomEffects Estimation Summary                           
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.7868
Estimator:                    RandomEffects   R-squared (Between):              0.8823
No. Observations:                       324   R-squared (Within):               0.5231
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.7868
Time:                              23:10:48   Log-likelihood                   -709.89
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.2393
Entities:                                 3   P-value                           0.0000
Avg Obs:                             108.00   Distribution:                 F(172,151)
Min Obs:                             108.00                                           
Max Obs

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos aleatorios son preferibles.
                           RandomEffects Estimation Summary                           
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8055
Estimator:                    RandomEffects   R-squared (Between):              0.9017
No. Observations:                       321   R-squared (Within):               0.5388
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.8055
Time:                              23:10:49   Log-likelihood                   -689.32
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      3.3876
Entities:                                 3   P-value                           0.0000
Avg Obs:                             107.00   Distribution:                 F(176,144)
Min Obs:                             107.00                                           
Max Obs

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -51.24543906939711
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8736
Estimator:                         PanelOLS   R-squared (Between):              0.2727
No. Observations:                       318   R-squared (Within):               0.8736
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.4324
Time:                              23:10:49   Log-likelihood                   -403.90
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.1844
Entities:                                 3   P-value                           0.0000
Avg Obs:                             106.00   Distribution:                 F(180,135)
Min Obs:                             106.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -99.43571299126742
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8818
Estimator:                         PanelOLS   R-squared (Between):              0.2813
No. Observations:                       315   R-squared (Within):               0.8818
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.4389
Time:                              23:10:50   Log-likelihood                   -388.70
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.1909
Entities:                                 3   P-value                           0.0000
Avg Obs:                             105.00   Distribution:                 F(184,128)
Min Obs:                             105.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -152.7379668269911
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8883
Estimator:                         PanelOLS   R-squared (Between):              0.2956
No. Observations:                       312   R-squared (Within):               0.8883
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.4496
Time:                              23:10:51   Log-likelihood                   -375.63
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.1200
Entities:                                 3   P-value                           0.0000
Avg Obs:                             104.00   Distribution:                 F(188,121)
Min Obs:                             104.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -256.654187173821
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.8957
Estimator:                         PanelOLS   R-squared (Between):              0.3579
No. Observations:                       309   R-squared (Within):               0.8957
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.4963
Time:                              23:10:51   Log-likelihood                   -361.12
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      5.0975
Entities:                                 3   P-value                           0.0000
Avg Obs:                             103.00   Distribution:                 F(192,114)
Min Obs:                             103.00                                

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -468.6896014254184
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.9004
Estimator:                         PanelOLS   R-squared (Between):              0.3773
No. Observations:                       306   R-squared (Within):               0.9004
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.5102
Time:                              23:10:52   Log-likelihood                   -349.39
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.9379
Entities:                                 3   P-value                           0.0000
Avg Obs:                             102.00   Distribution:                 F(196,107)
Min Obs:                             102.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


chi2: -1613.611573629349
Efectos fijos son preferibles.
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.9060
Estimator:                         PanelOLS   R-squared (Between):              0.4124
No. Observations:                       303   R-squared (Within):               0.9060
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.5355
Time:                              23:10:53   Log-likelihood                   -335.39
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.8189
Entities:                                 3   P-value                           0.0000
Avg Obs:                             101.00   Distribution:                 F(200,100)
Min Obs:                             101.00                               

/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:2623: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)
/usr/local/lib/python3.10/dist-packages/linearmodels/panel/model.py:1219: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  super().__init__(dependent, exog, weights=weights, check_rank=check_rank)


Efectos fijos son preferibles (p-valor < 0.05).
                             PanelOLS Estimation Summary                              
Dep. Variable:     Peso_total_del_racimo_kg   R-squared:                        0.9123
Estimator:                         PanelOLS   R-squared (Between):              0.4914
No. Observations:                       300   R-squared (Within):               0.9123
Date:                      Fri, Jun 14 2024   R-squared (Overall):              0.5945
Time:                              23:10:54   Log-likelihood                   -320.02
Cov. Estimator:                  Unadjusted                                           
                                              F-statistic:                      4.7405
Entities:                                 3   P-value                           0.0000
Avg Obs:                            100.000   Distribution:                  F(204,93)
Min Obs:                            100.000                                       

In [13]:
results = [model["result"] for model in models]
stargazer = Stargazer(results)

# Collect metrics for all models
rmse_values = [f"{np.sqrt(np.mean(model['result'].resids ** 2)):.4f}" for model in models]
mae_values = [f"{np.mean(np.abs(model['result'].resids)):.4f}" for model in models]
aic_values = [f"{model['aic']:.4f}" for model in models]
bic_values = [f"{model['bic']:.4f}" for model in models]
hqic_values = [f"{model['hqic']:.4f}" for model in models]

# Add lines to Stargazer with lists of values
stargazer.add_line("RMSE", rmse_values)
stargazer.add_line("MAE", mae_values)
stargazer.add_line("AIC", aic_values)
stargazer.add_line("BIC", bic_values)
stargazer.add_line("HQIC", hqic_values)

stargazer  # Display the table

In [14]:
lista_modelos['conteo_variables_predictoras'] = lista_modelos['variables_predictoras'].apply(lambda x: len(x))
lista_modelos['tipo_efectos'].value_counts()

tipo_efectos
Efectos fijos         48
Efectos Aleatorios     3
Name: count, dtype: int64

In [15]:
lista_modelos['tipo_efectos'].value_counts('%')

tipo_efectos
Efectos fijos         0.941176
Efectos Aleatorios    0.058824
Name: proportion, dtype: float64

In [16]:
lista_modelos

variables_predictoras  \
0   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
1   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
2   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
3   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
4   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
5   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
6   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
7   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
8   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
9   [temp_media_lag1, Precipitacion_lag1, Humedad_...   
10  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
11  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
12  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
13  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
14  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
15  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
16  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
17  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
18  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
19  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
20  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
21  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
22  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
23  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
24  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
25  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
26  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
27  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
28  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
29  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
30  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
31  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
32  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
33  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
34  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
35  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
36  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
37  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
38  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
39  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
40  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
41  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
42  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
43  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
44  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
45  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
46  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
47  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
48  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
49  [temp_media_lag1, Precipitacion_lag1, Humedad_...   
50  [temp_media_lag1, Precipitacion_lag1, Humedad_...   

                                               modelo        tipo_efectos  \
0                                PanelOLS Estimati...       Efectos fijos   
1                                PanelOLS Estimati...       Efectos fijos   
2                                PanelOLS Estimati...       Efectos fijos   
3                                PanelOLS Estimati...       Efectos fijos   
4                                PanelOLS Estimati...       Efectos fijos   
5                                PanelOLS Estimati...       Efectos fijos   
6                                PanelOLS Estimati...       Efectos fijos   
7                                PanelOLS Estimati...       Efectos fijos   
8                                PanelOLS Estimati...       Efectos fijos   
9                                PanelOLS Estimati...       Efectos fijos   
10                               PanelOLS Estimati...       Efectos fijos   
11                               PanelOLS Estimati...       Efectos fijos   
12                               PanelOLS Estimati...       Efecto

Vemos que para el 94% de los modelos entrenados en funcion de los rezagos responden a un efecto fijo, por lo cual se agregaran las variables dummies por finca para que ahi se recoga dicho efecto en los interceptos